File: java/milky doughnut/TaskSeven.java

public class TaskSeven{
public static int[] intersectOf(int[] numbers1, int[] numbers2) {

    int[] temp = new int[numbers1.length];
    int count = 0;

    for (int index = 0; index < numbers1.length; index++) {

        for (int position = 0; position < numbers2.length; position++) {

            if (numbers1[index] == numbers2[position]) {

                boolean alreadyExists = false;

                for (int counter = 0; counter < count; counter++) {
                    if (temp[counter] == numbers1[index]) {
                        alreadyExists = true;
                        break;
                    }
                }

                if (!alreadyExists) {
                    temp[count] = numbers1[index];
                    count++;
                }

                break;
            }
        }
    }

    int[] result = new int[count];

    for (int counter = 0; counter < count; counter++) {
        result[counter] = temp[counter];
    }

    return result;
}

}

File: python/milky doughnut/task_seven.py
def intersection(numbers1, numbers2):
    temp = []
    
    for index in range(len(numbers1)):
        for position in range(len(numbers2)):
            if numbers1[index] == numbers2[position]:
                
                
                already_exists = False
                for counter in range(len(temp)):
                    if temp[counter] == numbers1[index]:
                        already_exists = True
                        break
                
                if not already_exists:
                    temp.append(numbers1[index])
                
                break  
    
    return temp



